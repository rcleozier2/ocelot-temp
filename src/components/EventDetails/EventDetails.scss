.icon-green {
    color: green;
}

.attachments img {
    max-width: 70% !important;
    max-height: 300px;
    height: auto;
    clear: both;
}