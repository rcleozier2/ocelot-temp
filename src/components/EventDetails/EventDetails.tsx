import React from "react";
import { isNull } from "lodash";
import { format } from "date-fns";
import "./EventDetails.scss";

const EventDetails = (props: any) => {
  const normalizedTask = props.normalizedTask;

  const actualArrivalTime = !isNull(normalizedTask.times.actualArrivalTime)
    ? format(
        new Date(normalizedTask.times.actualArrivalTime),
        "MM/dd/yyyy h:mm a"
      )
    : "Not Set";

  const actualCompletionTime = !isNull(
    normalizedTask.times.actualCompletionTime
  )
    ? format(
        new Date(normalizedTask.times.actualCompletionTime),
        "MM/dd/yyyy h:mm a"
      )
    : "Not Set";

  const estimatedArrivalTime = !isNull(
    normalizedTask.times.estimatedArrivalTime
  )
    ? format(
        new Date(normalizedTask.times.estimatedArrivalTime),
        "MM/dd/yyyy h:mm a"
      )
    : "Not Set";

  const completeAfter = !isNull(normalizedTask.times.completeAfter)
    ? format(new Date(normalizedTask.times.completeAfter), "MM/dd/yyyy h:mm a")
    : "Not Set";

  const completeBefore = !isNull(normalizedTask.times.completeBefore)
    ? format(new Date(normalizedTask.times.completeBefore), "MM/dd/yyyy h:mm a")
    : "Not Set";

  return (
    <>
      <p className="summary-container-title">
        {normalizedTask.failed === true && (
          <span>
            <i className="pi pi-exclamation-triangle icon icon-red"></i>
            Failed
          </span>
        )}
      </p>

      <p className="summary-text m-0">
        <i className="pi pi-directions icon-small"></i> <span>Driver: </span>
        {normalizedTask.driver}
      </p>

      <p className="summary-text m-0">
        <i className="pi pi-user icon-small"></i> <span>Admin: </span>
        {normalizedTask.admin}
      </p>

      <hr />
      <p className="summary-text m-0">
        {normalizedTask.recipients[0].name} <br />
        {normalizedTask.address.number} {normalizedTask.address.street} <br />
        {normalizedTask.address.city}, {normalizedTask.address.state}{" "}
        {normalizedTask.address.postalCode} <br />
        {normalizedTask.recipients[0].phone} <br />
      </p>

      {/* <p className="summary-text m-0">
            <span>Total time to completion:</span>
            {normalizedTask.times.totalTime}
          </p> */}

      <hr />
      <p className="summary-text m-0">
        <i className="pi pi-clock icon-small"></i> <span>Delivery Window </span>
        {completeAfter} - {completeBefore}
        {normalizedTask.times.completedStatus !== "ontime" && (
          <p className="summary-notes-time m-0">
            {normalizedTask.times.completedStatus} by{" "}
            {normalizedTask.times.completionTimeDifference} Minutes{" "}
          </p>
        )}
      </p>
      <hr />

      <p className="summary-text m-0">
        <i className="pi pi-clock icon-small"></i>{" "}
        <span>Estimated Arrival Time:</span>
        {estimatedArrivalTime}
      </p>

      <p className="summary-text m-0">
        <i className="pi pi-clock icon-small"></i>{" "}
        <span>Actual Arrival Time:</span>
        {actualArrivalTime}
      </p>

      {normalizedTask.times.arrivalStatus !== "ontime" && (
        <p className="summary-notes-notice m-0">
          {normalizedTask.times.arrivalStatus} by Onfleets estimate by{" "}
          {normalizedTask.times.arrivalTimeDifference} Minutes{" "}
        </p>
      )}

      <hr />

      <p className="summary-text m-0">
        <i className="pi pi-clock icon-small"></i>{" "}
        <span>Actual Completed Time:</span>
        {actualCompletionTime}
      </p>

      <hr />

      {normalizedTask.notes.task !== "" && (
        <div className="summary-notes">
          <p>Task Notes:</p>
          <span className="className">{normalizedTask.notes.task}</span>
        </div>
      )}

      {normalizedTask.notes.recipients !== "" && (
        <div className="summary-notes">
          <p>Recipient Notes:</p>
          <span className="className">{normalizedTask.notes.recipients}</span>
        </div>
      )}

      {normalizedTask.notes.driver !== "" && (
        <div className="summary-notes">
          <p>Driver Notes: </p>
          <span> {normalizedTask.notes.driver}</span>
        </div>
      )}
    </>
  );
};

export default EventDetails;
