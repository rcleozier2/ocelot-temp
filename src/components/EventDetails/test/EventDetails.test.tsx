import React from "react";
import EventDetails from "../EventDetails";
import renderer from "react-test-renderer";

xit("renders EventDetails correctly for completed task", () => {
  const props = {
    normalizedTask : {
        times : {
            actualArrivalTime: 1575574061,
            actualCompletionTime: 1575574062,
            estimatedArrivalTime: 1575574063,
            completeAfter: 1575574061,
            completeBefore: 1575574062,
            completedStatus: true,
            completionTimeDifference: 23,
            arrivalStatus: true,
            arrivalTimeDifference: 10

        },
        failed: false,
        completed: true,
        driver: "Driver 1",
        admin: "Medly Admin",
        recipients: [{
            name: "Name",
            phone: "+212-122-2232"
        }],
        address:{
            number : 1,
            street: "110 Street",
            city: "Brooklyn",
            state: "New York",
            postalCode: "11101"
        },
        notes: {
            task: "task notes",
            recipients: "recipients note",
            driver: "driver notes"

        },
        attachments: []
    }
  };

  const component = renderer.create(<EventDetails {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

xit("renders EventDetails correctly for failed task", () => {
    const props = {
      normalizedTask : {
          times : {
              actualArrivalTime: 1575574061,
              actualCompletionTime: 1575574062,
              estimatedArrivalTime: 1575574063,
              completeAfter: 1575574061,
              completeBefore: 1575574062,
              completedStatus: true,
              completionTimeDifference: 23,
              arrivalStatus: true,
              arrivalTimeDifference: 10
  
          },
          failed: true,
          completed: false,
          driver: "Driver 2",
          admin: "Medly Admin",
          recipients: [{
              name: "Name",
              phone: "+212-122-2232"
          }],
          address:{
              number : 1,
              street: "115 Street",
              city: "Brooklyn",
              state: "New York",
              postalCode: "11100"
          },
          notes: {
              task: "task notes",
              recipients: "recipients note",
              driver: "driver notes"
  
          },
          attachments: []
      }
    };
  
    const component = renderer.create(<EventDetails {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
