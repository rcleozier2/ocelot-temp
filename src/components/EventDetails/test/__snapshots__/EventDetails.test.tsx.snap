// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EventDetails correctly for completed task 1`] = `
Array [
  <p
    className="summary-container-title"
  >
    <span>
      <i
        className="pi pi-check icon icon-green"
      />
      Successful
    </span>
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-directions icon-small"
    />
     
    <span>
      Driver: 
    </span>
    Driver 1
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-user icon-small"
    />
     
    <span>
      Admin: 
    </span>
    Medly Admin
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    Name
     
    <br />
    1
     
    110 Street
     
    <br />
    Brooklyn
    , 
    New York
     
    11101
     
    <br />
    +212-122-2232
     
    <br />
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Delivery Window 
    </span>
    01/19/1970 12:39 AM
     - 
    01/19/1970 12:39 AM
    <p
      className="summary-notes-time m-0"
    >
       by
       
      23
       Minutes
       
    </p>
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Estimated Arrival Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Actual Arrival Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <p
    className="summary-notes-notice m-0"
  >
     by Onfleets estimate by
     
    10
     Minutes
     
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Actual Completed Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <hr />,
  <div
    className="summary-notes"
  >
    <p>
      Task Notes:
    </p>
    <span
      className="className"
    >
      task notes
    </span>
  </div>,
  <div
    className="summary-notes"
  >
    <p>
      Recipient Notes:
    </p>
    <span
      className="className"
    >
      recipients note
    </span>
  </div>,
  <div
    className="summary-notes"
  >
    <p>
      Driver Notes: 
    </p>
    <span>
       
      driver notes
    </span>
  </div>,
]
`;

exports[`renders EventDetails correctly for failed task 1`] = `
Array [
  <p
    className="summary-container-title"
  >
    <span>
      <i
        className="pi pi-exclamation-triangle icon icon-red"
      />
      Failed
    </span>
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-directions icon-small"
    />
     
    <span>
      Driver: 
    </span>
    Driver 2
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-user icon-small"
    />
     
    <span>
      Admin: 
    </span>
    Medly Admin
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    Name
     
    <br />
    1
     
    115 Street
     
    <br />
    Brooklyn
    , 
    New York
     
    11100
     
    <br />
    +212-122-2232
     
    <br />
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Delivery Window 
    </span>
    01/19/1970 12:39 AM
     - 
    01/19/1970 12:39 AM
    <p
      className="summary-notes-time m-0"
    >
       by
       
      23
       Minutes
       
    </p>
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Estimated Arrival Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Actual Arrival Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <p
    className="summary-notes-notice m-0"
  >
     by Onfleets estimate by
     
    10
     Minutes
     
  </p>,
  <hr />,
  <p
    className="summary-text m-0"
  >
    <i
      className="pi pi-clock icon-small"
    />
     
    <span>
      Actual Completed Time:
    </span>
    01/19/1970 12:39 AM
  </p>,
  <hr />,
  <div
    className="summary-notes"
  >
    <p>
      Task Notes:
    </p>
    <span
      className="className"
    >
      task notes
    </span>
  </div>,
  <div
    className="summary-notes"
  >
    <p>
      Recipient Notes:
    </p>
    <span
      className="className"
    >
      recipients note
    </span>
  </div>,
  <div
    className="summary-notes"
  >
    <p>
      Driver Notes: 
    </p>
    <span>
       
      driver notes
    </span>
  </div>,
]
`;
